controllers/imagerepository_controller.go

```diff
@@ -277,3 +301,51 @@ func authFromSecret(secret corev1.Secret, registry string) (authn.Authenticator,
 		return nil, fmt.Errorf("unknown secret type %q", secret.Type)
 	}
 }
+
+// event emits a Kubernetes event and forwards the event to notification controller if configured
+func (r *ImageRepositoryReconciler) event(repo imagev1alpha1.ImageRepository, severity, msg string) {
+	if r.EventRecorder != nil {
+		r.EventRecorder.Eventf(&repo, "Normal", severity, msg)
+	}
+	if r.ExternalEventRecorder != nil {
+		objRef, err := reference.GetReference(r.Scheme, &repo)
+		if err != nil {
+			r.Log.WithValues(
+				"request",
+				fmt.Sprintf("%s/%s", repo.GetNamespace(), repo.GetName()),
+			).Error(err, "unable to send event")
+			return
+		}
+
+		if err := r.ExternalEventRecorder.Eventf(*objRef, nil, severity, severity, msg); err != nil {
+			r.Log.WithValues(
+				"request",
+				fmt.Sprintf("%s/%s", repo.GetNamespace(), repo.GetName()),
+			).Error(err, "unable to send event")
+			return
+		}
+	}
+}
+
+func (r *ImageRepositoryReconciler) recordReadiness(repo imagev1alpha1.ImageRepository) {
```

Comment:
I didn't know what this method was for until I read through its code -- can we name it `recordReadinessMetric` instead?

Reply:
Makes sense to me. I suppose this is something worth applying to other controller components that define it.